File: result/translate/chatgpt/python2java/java/PROGRAM_BINARY_DECIMAL_CONVERSION_1.java
public static int f_gold(String num) {
    int dec_value = 0;
    int base1 = 1;
    int len1 = num.length();
    
    for (int i = len1 - 1; i >= 0; i--) {
        if (num.charAt(i) == '1') {
            dec_value += base1;
        }
        base1 = base1 * 2;
    }
    
    return dec_value;
}

File: result/translate/chatgpt/java2python/python/PROGRAM_BINARY_DECIMAL_CONVERSION_1.py
def f_gold(n):
    num = n
    dec_value = 0
    base = 1
    len_num = len(num)
    
    for i in range(len_num - 1, -1, -1):
        if num[i] == '1':
            dec_value += base
        base *= 2

    return dec_value